pxr/usd/lib/usdGeom/points.cpp

```diff
@@ -238,4 +238,82 @@ TF_REGISTRY_FUNCTION(UsdGeomBoundable)
         _ComputeExtentForPoints);
 }
 
+size_t
+UsdGeomPoints::ComputePositionsAtTimes(
+    std::vector<VtVec3fArray>& positions,
+    const std::vector<UsdTimeCode>& sampleTimes,
+    UsdTimeCode baseTime,
+    float velocityScale) const {
+    constexpr double epsilonTest = 1e-5;
+    const auto sampleCount = sampleTimes.size();
+    if (sampleCount == 0 || positions.size() < sampleCount || baseTime.IsDefault()) {
+        return 0;
+    }
+
+    const auto pointsAttr = GetPointsAttr();
+    if (!pointsAttr.HasValue()) {
+        return 0;
+    }
+
+    bool hasValue = false;
+    double pointsLowerTimeSample = 0.0;
+    double upperTimeSample = 0.0;
+    if (!pointsAttr.GetBracketingTimeSamples(baseTime.GetValue(), &pointsLowerTimeSample,
+                                             &upperTimeSample, &hasValue) || !hasValue) {
+        return 0;
+    }
+
+    VtVec3fArray points;
+    VtVec3fArray velocities;
+
+    // We need to check if there is a queriable velocity at the given point,
+    // if it has almost the same lower time sample and the array lengths are equal.
+    bool velocityExists = false;
+    const auto velocitiesAttr = GetVelocitiesAttr();
+    double velocitiesLowerTimeSample = 0.0;
+    if (velocitiesAttr.HasValue() &&
+        velocitiesAttr.GetBracketingTimeSamples(baseTime.GetValue(),
+                                                &velocitiesLowerTimeSample, &upperTimeSample, &hasValue)
+        && hasValue && GfIsClose(velocitiesLowerTimeSample, pointsLowerTimeSample, epsilonTest)) {
+        if (pointsAttr.Get(&points, pointsLowerTimeSample) &&
+            velocitiesAttr.Get(&velocities, pointsLowerTimeSample) &&
+            points.size() == velocities.size()) {
+            velocityExists = true;
+        }
+    }
+
+    if (velocityExists) {
+        const auto pointCount = points.size();
+        if (pointCount == 0) { return sampleCount; }
+        const auto timeCodesPerSecond = GetPrim().GetStage()->GetTimeCodesPerSecond();
+        for (auto a = decltype(sampleCount){0}; a < sampleCount; ++a) {
+            VtVec3fArray& curr = positions[a];
+            curr.resize(pointCount);
+            const auto currentMultiplier = static_cast<float>((sampleTimes[a].GetValue() - pointsLowerTimeSample)
+                                                              / timeCodesPerSecond) * velocityScale;
+            for (auto i = decltype(pointCount){0}; i < pointCount; ++i) {
+                curr[i] = points[i] + velocities[i] * currentMultiplier;
+            }
+        }
+        return sampleCount;
+    } else {
+        if (!pointsAttr.Get(&positions[0], sampleTimes[0])) { return 0; }
+
+        const auto idsAttr = GetIdsAttr();
+        VtArray<long> ids;
+        if (!idsAttr.Get(&ids, sampleTimes[0])) { return 1; }
```

Comment:
Sure. I'll remove the ID checks for now, and just use the built-in interpolation for points. I'll check for point lengths though. Also, this means I can move the function back to pointBased. (my original implementation was there, but I promoted it to points because of the extra id checks)